### PR TITLE
Fix Issue 18230 - multiwayUnion sets wrong pred lambdas

### PR DESCRIPTION
**Pull request by RazvanN7**
cc @9il @wilzbach

### CODE REVIEW

File: std/algorithm/setops.d [hunk 1]
```diff
@@ -949,7 +949,8 @@ See also: $(LREF multiwayMerge)
 auto multiwayUnion(alias less = "a < b", RangeOfRanges)(RangeOfRanges ror)
 {
     import std.algorithm.iteration : uniq;
-    return ror.multiwayMerge.uniq;
+    import std.functional : not;
+    return multiwayMerge!less(ror).uniq!(not!less);
```
Comment (JackStouffer) [inline std/algorithm/setops.d]: nitpick: please don't mix non-UFCS and UFCS.

Either `ror.multiwayMerge!less.uniq!(not!less)` or `uniq!(not!less)(multiwayMerge!less(ror))`.

Comment (JackStouffer) [inline std/algorithm/setops.d]: The pred use for `uniq` doesn't seem correct. `uniq`'s pred is supposed to be for equality

File: std/algorithm/setops.d [hunk 2]
```diff
@@ -980,6 +981,15 @@ auto multiwayUnion(alias less = "a < b", RangeOfRanges)(RangeOfRanges ror)
         [ 7 ],
     ];
     assert(equal(multiwayUnion(b), witness));
+
+    double[][] c =
+    [
+        [9, 8, 7, 6],
```
Comment (wilzbach) [inline std/algorithm/setops.d]: Add: `8, 8, 8` here

File: std/algorithm/setops.d (hunk 1 above)
Comment (9il) [reply std/algorithm/setops.d]: It is correct here for current uniq implementation. And it is a big task to invent a uniq implementation that will not work here. The `not!less` solutions seems to be the best here

**Comment by 9il** [comment]
LGTM

**Pull request merged**